openapi-comparator-tests/src/test/scala/sttp/apispec/openapi/validation/OpenAPIComparatorTest.scala

```diff
@@ -0,0 +1,395 @@
+package sttp.apispec.openapi.validation
+
+import org.scalatest.funsuite.AnyFunSuite
+import sttp.apispec.SchemaType
+import sttp.apispec.openapi.{OpenAPI, ResponsesCodeKey}
+import sttp.apispec.openapi.circe.openAPIDecoder
+import sttp.apispec.test.ResourcePlatform
+import sttp.apispec.validation.TypeMismatch
+
+class OpenAPIComparatorTest extends AnyFunSuite with ResourcePlatform {
+  override val basedir = "openapi-comparator-tests"
+
+  test("identical") {
+    val Right(clientOpenapi) = readJson("/petstore/basic-petstore-v_1_2_4.json").flatMap(_.as[OpenAPI]): @unchecked
+    val Right(serverOpenapi) = readJson("/petstore/basic-petstore-v_1_2_4.json").flatMap(_.as[OpenAPI]): @unchecked
+
+    val openAPIComparator = new OpenAPIComparator(clientOpenapi, serverOpenapi)
+
+    assert(openAPIComparator.compare().isEmpty)
+  }
+
+  test("missing path") {
+    val Right(clientOpenapi) = readJson("/petstore/basic-petstore-v_1_0_2.json").flatMap(_.as[OpenAPI]): @unchecked
```

Comment:
Another suggestion: the v_1_0_0.json filenames are very opaque. Maybe it would make sense to have them per-test-case, that is `petstore-missing-path-client.json` and `petstore-missing-path-server.json`

or even: `missing-path-client.json` and `missing-path-server.json`. Then they're grouped into pairs which are used in tests, and it's immediately clear what's the purpose of a file